activesupport/lib/active_support/message_encryptor.rb

```diff
@@ -43,65 +50,181 @@ class InvalidMessage < StandardError; end
     # * <tt>:digest</tt> - String of digest to use for signing. Default is +SHA1+.
     # * <tt>:serializer</tt> - Object serializer to use. Default is +Marshal+.
     def initialize(secret, *signature_key_or_options)
-      options = signature_key_or_options.extract_options!
-      sign_secret = signature_key_or_options.first
+      @options = signature_key_or_options.extract_options!
       @secret = secret
-      @sign_secret = sign_secret
-      @cipher = options[:cipher] || 'aes-256-cbc'
```

Comment:
I don't understand how :sweat_smile: . Can you please explain ? In my understanding, backwards compatibility means that all the previous Tokens should be handled by our implementation. This is taken care by LegacyEncryptor https://github.com/sbhatore/rails/pull/4/files#diff-3aeeced39aff3bbba1a5c38d367eab49R179